cmd/kube-controller-manager/app/options/options.go

```diff
@@ -131,6 +131,7 @@ func (s *CMServer) AddFlags(fs *pflag.FlagSet, allControllers []string, disabled
 	fs.BoolVar(&s.UseServiceAccountCredentials, "use-service-account-credentials", s.UseServiceAccountCredentials, "If true, use individual service account credentials for each controller.")
 	fs.StringVar(&s.CloudProvider, "cloud-provider", s.CloudProvider, "The provider for cloud services.  Empty string for no provider.")
 	fs.StringVar(&s.CloudConfigFile, "cloud-config", s.CloudConfigFile, "The path to the cloud provider configuration file.  Empty string for no configuration file.")
+	fs.StringVar(&s.ExternalPlugin, "external-plugin", s.ExternalPlugin, "The plugin to use when cloud provider external. Can be empty, should only be set when cloud-provider is external.")
```

Comment:
Volume plugin to use when Cloud provider is external. Eg. 
`--external-plugin=aws` (for ebs )
`--external-plugin=gce` (for pd) etc.

Reply:
Shouldn't the name say "volume" if it is about volumes?  But it's not a volume plugin, it's a cloud plugin.  Can we just make this extra extra clear?

Reply:
I was very confused as well. What @thockin said.